flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java

```diff
@@ -0,0 +1,235 @@
+                    .addSchema(
+                            "ts bigint",
+                            "a bigint",
+                            "b int",
+                            "c string",
+                            "rowtime as TO_TIMESTAMP(FROM_UNIXTIME(ts))",
+                            "watermark for rowtime as rowtime")
+                    .producedBeforeRestore(DATA)
+                    .producedAfterRestore(AFTER_DATA)
+                    .build();
+
+    public static final String[] BEFORE_RESTORE_DATA = {
+        "+I[Hello, 10, 1970-01-01T00:00:10, 1, 0, 1, 1]",
+        "+I[Hello, 15, 1970-01-01T00:00:15, 15, 0, 2, 2]",
+        "+I[Hello, 16, 1970-01-01T00:00:16, 16, 0, 3, 3]",
+        "+I[Hello, 20, 1970-01-01T00:00:20, 2, 0, 4, 5]",
+        "+I[Hello, 30, 1970-01-01T00:00:30, 3, 0, 2, 5]",
+        "+I[Hello, 40, 1970-01-01T00:00:40, 4, 0, 2, 7]",
+        "+I[Hello, 50, 1970-01-01T00:00:50, 5, 1, 2, 9]",
+        "+I[Hello, 60, 1970-01-01T00:01, 6, 2, 2, 11]",
+        "+I[Hello, 65, 1970-01-01T00:01:05, 65, 2, 2, 12]",
+        "+I[Hello, 90, 1970-01-01T00:01:30, 9, 1, 1, 6]",
+        "+I[Hello, 95, 1970-01-01T00:01:35, 18, 2, 2, 12]",
+        "+I[Hello World, 100, 1970-01-01T00:01:40, 7, 1, 1, 7]",
+        "+I[Hello, 99, 1970-01-01T00:01:39, 15, 3, 3, 31]",
+        "+I[Hello World, 110, 1970-01-01T00:01:50, 17, 2, 2, 14]",
+        "+I[Hello World, 140, 1970-01-01T00:02:20, 18, 1, 1, 7]",
+        "+I[Hello, 53, 1970-01-01T00:00:53, 15, 1, 1, 19]",
+        "+I[Hello World, 150, 1970-01-01T00:02:30, 8, 2, 2, 15]",
+        "+I[Hello World, 200, 1970-01-01T00:03:20, 20, 1, 1, 20]"
+    };
+    public static final String[] AFTER_RESTORE_OUTPUT = {
+        "+I[Hello, 13, 1970-01-01T00:00:13, 1, 0, 1, 1]",
+        "+I[Hello, 19, 1970-01-01T00:00:19, 15, 0, 2, 2]",
+        "+I[Hello, 33, 1970-01-01T00:00:33, 1, 1, 1, 17]",
+        "+I[Hello, 37, 1970-01-01T00:00:37, 15, 2, 2, 36]",
+        "+I[Hello, 52, 1970-01-01T00:00:52, 15, 1, 1, 19]",
+        "+I[Hello World, 202, 1970-01-01T00:03:22, 20, 2, 2, 40]"
+    };
+
+    static final TableTestProgram OVER_AGGREGATE_BOUNDED_PARTITIONED_ROWS =
+            getTableTestProgram(
+                    "over-aggregate-bounded-partitioned-rows",
+                    "validates over aggregate node",
+                    "PARTITION BY c ORDER BY rowtime RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW",
+                    BEFORE_RESTORE_DATA,
+                    AFTER_RESTORE_OUTPUT);
+
+    static final TableTestProgram OVER_AGGREGATE_BOUNDED_NON_PARTITIONED_ROWS =
+            getTableTestProgram(
+                    "over-aggregate-bounded-non-partitioned-rows",
+                    "validates over aggregate node",
+                    "PARTITION BY c ORDER BY rowtime RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW",
+                    BEFORE_RESTORE_DATA,
+                    AFTER_RESTORE_OUTPUT);
```

Comment:
Are those two different? I can't see a difference in queries. Could you also update the descriptions, they're also the same now.